eng/scripts/Get-PackageVersions.ps1

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env pwsh
+
+#Requires -Version 7.0
+[CmdletBinding()]
+param(
+  [switch]$Dependencies
+)
+
+$ErrorActionPreference = 'Stop'
+
+. (Join-Path $PSScriptRoot '..' 'common' 'scripts' 'common.ps1')
+
+$metadata = cargo metadata --format-version 1 --no-deps --all-features | ConvertFrom-Json -AsHashtable
+$packages = $metadata.packages
+foreach ($package in $packages) {
+  try {
+    $name = $package.name
+    $resp = Invoke-WebRequest "https://index.crates.io/$($name.Substring(0,2))/$($name.Substring(2,2))/$name"
+    $packageVersions = $resp.Content.Trim().Split("`n") | ConvertFrom-Json | Select-Object -ExpandProperty vers
+    $package.indexVersion = $packageVersions | Sort-Object { [AzureEngSemanticVersion]::ParseVersionString($_) } | Select-Object -Last 1
+  }
+  catch {
+  }
```

Comment:
Are you sure you don't want to skip, or is the idea here that it's not published so treat it new? This would mean the `version` and `indexVersion` down below on lines 32, 33 currently would be `$null`. From this file along I can't tell what the downstream ramifications of that are, though.

Reply:
good point.  I added explicit 404 handling and $null setting in the catch block.  This section doesn't set `version`